src/installer/corehost/corehost.cpp

```diff
@@ -200,23 +187,32 @@ int exe_start(const int argc, const pal::char_t* argv[])
 
     // Obtain the entrypoints.
     int rc;
-    hostfxr_main_startupinfo_fn main_fn_v2 = reinterpret_cast<hostfxr_main_startupinfo_fn>(pal::get_symbol(fxr, "hostfxr_main_startupinfo"));
-    if (main_fn_v2 != nullptr)
+#if defined(FEATURE_APPHOST)
+    if (bundle_marker_t::is_bundle())
     {
+        hostfxr_main_bundle_startup_info_fn main_fn_v3 = reinterpret_cast<hostfxr_main_bundle_startup_info_fn>(pal::get_symbol(fxr, "hostfxr_main_bundle_startup_info"));
+        if (main_fn_v3 == nullptr)
+        {
+            trace::error(_X("The required library %s does not support single-file apps."), fxr_path.c_str());
+            rc = StatusCode::CoreHostEntryPointFailure;
+        }
+
         const pal::char_t* host_path_cstr = host_path.c_str();
         const pal::char_t* dotnet_root_cstr = dotnet_root.empty() ? nullptr : dotnet_root.c_str();
         const pal::char_t* app_path_cstr = app_path.empty() ? nullptr : app_path.c_str();
+        int64_t bundle_header_offset = bundle_marker_t::header_offset();
 
-        trace::info(_X("Invoking fx resolver [%s] v2"), fxr_path.c_str());
+        trace::info(_X("Invoking fx resolver [%s] v3"), fxr_path.c_str());
```

Comment:
What is the "v3" for? It doesn't really match any version - and the entry points are not versioned in their name. Can we please remove this notion - it's just confusing.
Let's just trace the name of the entry point we're calling. (same below).
Could you also please rename the local variables to not refer to versions...